presto-docs/src/main/sphinx/develop/delete-and-update.rst

```diff
@@ -0,0 +1,178 @@
+====================================
+Supporting ``DELETE`` and ``UPDATE``
+====================================
+
+The Presto engine provides APIs to support row-level SQL ``DELETE`` and ``UPDATE``.
+To implement ``DELETE`` or ``UPDATE``, a connector must layer an ``UpdatablePageSource``
+on top of the connector's usual ``ConnectorPageSource``, and define ``ConnectorMetadata``
+methods to get a "rowId" column handle; to start the operation; and to finish the operation.
+
+``DELETE`` and ``UPDATE`` Data Flow
+===================================
+
+``DELETE`` and ``UPDATE`` have a similar flow:
+
+* For each split, the connector will create an ``UpdatablePageSource`` instance, layered over the
+  connector's ``ConnectorPageSource``, to read pages on behalf of the Presto engine, and to
+  write deletions and/or updates to the underlying data store.
+* The connector's ``UpdatablePageSource.getNextPage()`` implementation fetches the next page
+  from the underlying ``ConnectorPageSource``, optionally reformats the page, and returns it
+  to the Presto engine.
+* The Presto engine performs filtering and projection on the page read, producing a page of filtered,
+  projected results.
+* The Presto engine passes that filtered, projected page of results to the connector's
+  ``UpdatablePageSource`` ``deleteRows()`` or ``updateRows()`` method.  Those methods persist
+  the deletions or updates in the underlying data store.
+* When all the pages for a specific split have been processed, the Presto engine calls
+  ``UpdatablePageSource.finish()``, which returns a ``Collection<Slice>`` of "fragments"
+  representing connector-specific information about the rows processed by the calls to
+  ``deleteRows`` or ``updateRows``.
+* When all pages for all splits have been processed, the Presto engine calls ``ConnectorMetadata.finishDelete()`` or
+  ``finishUpdate``, passing a collection containing all the "fragments" from all the splits.  The connector
+  does what is required to finalize the operation, for example, committing the transaction.
+
+The rowId Column Abstraction
```

Comment:
Consider adding "handle" to the heading titles in lines 34, 41, and 52, to help the reader find the explanation of this concept.